packages/transformer/src/IModelTransformer.ts

```diff
@@ -1240,6 +1248,10 @@ export class IModelTransformer extends IModelExportHandler {
     return targetRelationshipProps;
   }
 
+  public override shouldExportElementAspect(aspect: ElementAspect) {
+    return this.context.findTargetElementId(aspect.element.id) !== Id64.invalid;
```

Comment:
This seems incorrect... `shouldExportElementAspect` should be based on the aspect itself, not the transformer's internal processing state of that aspect. Is it possible to remove this requirement?

Reply:
Hardly, the issue is that detached aspect exporting will also export aspects for elements which are not exported. This check is for checking if aspect's element was actually exported. The only way I see that this could be moved to `IModelExporter` if it would track which elements were not exported and that list would be used to check if aspect should be exported in detached aspect processing. I'll try to look if there is another approach to this problem.

Reply:
yes we would need to know the list of exported elements since shouldExportElement is not sufficient if they call process/exportElement themselves... I think I'm OK with what you have here now, so long as you add a comment explaining this issue.

Reply:
Added comment, and also specified under `DetachedExportElementAspectsStrategy` that it will export all aspects regardless if the element was exported or not and that it needs to be handled by the consuming classes.